src/debug.cc

```diff
@@ -421,24 +421,25 @@ syslog_facility_names[] = {
     }
 };
 
-#endif
+#endif /* LOG_LOCAL4 */
+#endif /* HAVE_SYSLOG */
 
 void
 _db_set_syslog(const char *facility)
 {
+#if !HAVE_SYSLOG
+    fatal("Logging to syslog not available on this platform");
```

Comment:
I wouldn't expect it to make much of a difference, also because it's really hard to think of a platform without syslog support anymore. Maybe windows, and it's probably emulated there as well.

Reply:
... and systemd machines. There is a minor fight going on between journald and syslog proponents recently. The way its going we are going to have to add a module someday soon that drops log lines onto the systemd data bus.

Reply:
> Alex: Please try throwing an TextException instead. Does it give an overall better output for the user?

> Francesco: I wouldn't expect it to make much of a difference

I am not asking because I want to waste your time or want to replace one good solution with another good solution. I have a reason to believe that an exception may offer better UX in this specific use case (besides being a better error handling mechanism than fatal() in general).

BTW, _before_ trying an exception: Please remove the duplication of this refactored check/functionality in main.cc. That file should call this function unconditionally.

Reply:
It's a trivial change, so sure. For the future, it'd be good if you would explain your reasoning before making similar requests, as this threads is structured it feels like nit picking. Code should win over opinions, and this is not holding true in this case.
The change to main.cc has already happened; I had noticed this double check earlier on.
Fun thing, in main.cc we have some unconditional calls to syslog, which is emulated on windows but not on Unix; in other words right now unless I'm mistaken we would fail to build on any platform that lacks providing the syslog library call

Reply:
> it'd be good if you would explain your reasoning before making similar requests

I apologize. FWIW, I always try to mark non-obvious nits explicitly and provide reasoning behind other non-obvious change requests. I should not have done that in the form of a question in this change request. I became sloppy after investing so much time into cleaning up this PR. Sorry.

> Code should win over opinions, and this is not holding true in this case.

Sorry, I do not understand how that phrase applies to (or what it means in the context of) this change request.